linode/manager
# feat: [M3-7343] – Display conditional Notices depending on config scenarios in Linode Add/Edit Config dialog

## Description 📝
If a user is about to select an unrecommended configuration in the Add/Edit Config dialog, we want to display conditional warning notices informing them of this and what we recommend to avoid the network connectivity issues that would come with the unrecommended configuration.

## Changes  🔄
- Added `unrecommendedConfigNoticeSelector` function
- Added several constants for copy
- Display conditional notices depending on scenario in Add/Edit Config dialog

## Preview 📷
<details>
<summary>Scenario 1</summary>

- Have a VPC interface and another non-VPC interface (`public` is fine)
- the index of the primary interface !== the index of the VPC interface (i.e., the non-VPC interface should be primary)
- `nattedIPv4Address` (i.e., "Assign a public IPv4 address for this Linode" checked)

![Screenshot 2023-11-20 at 10 51 31 AM](https://github.com/linode/manager/assets/114682940/238adcfa-f256-4680-8071-dda1a909d92d)
</details>

<details>
<summary>Scenario 2</summary>

- all of Scenario 1, except: `!nattedIPv4Address` (i.e., "Assign a public IPv4 address for this Linode" unchecked)

![Screenshot 2023-11-20 at 10 50 43 AM](https://github.com/linode/manager/assets/114682940/bd956a9f-500e-40e2-a6c5-d62c19333df3)
</details>

<details>
<summary>Scenario 3</summary>

- only eth0 populated, and it is a VPC interface where `!nattedIPv4Address` (i.e., "Assign a public IPv4 address for this Linode" unchecked)

![Screenshot 2023-11-20 at 10 52 23 AM](https://github.com/linode/manager/assets/114682940/6b837ae0-9a1e-4955-a8a1-c2c810375b0f)
</details>

## How to test 🧪

### Prerequisites
- An account with the proper VPC tags

### Verification steps 
In the Add/Edit Config dialog, ensure that the specific scenarios outlined in the `Preview` section above are all handled as expected, and that none of the notices display outside of the outlined scenarios.

## As an Author I have considered 🤔
